### PR TITLE
Improve matching of segments against CFI ranges

### PR DESCRIPTION
**Pull request by robertknight**
This PR contains a couple of improvements to how we match CFI ranges (eg. for ebook assignments) against the CFI of the current book segment:

1. Ignore the part of CFIs after the step indirection ("!") (first commit). As a reminder, the part before the step indirection identifies the content document/HTML file, the part after identifies the location within that file.
2. Ignore hyphens inside assertions when splitting CFI ranges (second commit)

This resolves an issue where the user was incorrectly presented with a "you are outside the assignment" notice when a VitalSource assignment was created, and the CFIs in the table of contents had step indirections in them. See the added test case in `annotation-segment-test.js` for an example. See also [this Slack thread](https://hypothes-is.slack.com/archives/C4K6M7P5E/p1710190672323349) reporting the issue.

### CODE REVIEW

File: src/shared/cfi.ts [hunk 1]
```diff
@@ -34,6 +34,28 @@ function compareArrays(
...
+export function splitCFIRange(range: string): [string, string] {
+  const rangeWithoutAssertions = stripCFIAssertions(range);
+
+  // eslint-disable-next-line prefer-const
+  let [start, end] = rangeWithoutAssertions.split('-', 2);
+  if (end === undefined) {
+    end = start;
+  }
+
+  return [start, end];
```
Comment (acelaya) [inline src/shared/cfi.ts:56]: You could use `const` here by setting `start` as the default value for `end`:

```suggestion
  const [start, end = start] = rangeWithoutAssertions.split('-', 2);
  return [start, end];
```

**Review by acelaya — APPROVED** [review]
Add a small suggestion for `splitCFIRange`. Other than that, looks good.

File: src/shared/cfi.ts (hunk 1 above)
Comment (acelaya) [reply src/shared/cfi.ts:56]: Alternatively, you can do `const [start, end = start] = ...;` and `return [start, end ?? start];` if you think it's more clear.

Comment (robertknight) [reply src/shared/cfi.ts]: The second option sounds good. I'll do that.

**Pull request merged**